src/OpenFeature/Api.cs

```diff
@@ -127,7 +127,26 @@ public FeatureProvider GetProvider(string clientName)
         /// </para>
         /// </summary>
         /// <param name="hooks">A list of <see cref="Hook"/></param>
-        public void AddHooks(IEnumerable<Hook> hooks) => this._hooks.PushRange(hooks.ToArray());
+        public void AddHooks(IEnumerable<Hook> hooks)
+#if NET7_0_OR_GREATER
+            => this._hooks.PushRange(hooks as Hook[] ?? hooks.ToArray());

```

Comment:
I think we may have the same issue with the `client`: https://github.com/open-feature/dotnet-sdk/blob/a6062fe2b9f0d83490c7ce900e837863521f5f55/src/OpenFeature/OpenFeatureClient.cs#L109